From 316e6c0337d7f439d8e4f0f9d8215d09bc163517 Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Wed, 17 Jan 2024 13:07:20 +0530
Subject: [PATCH] wifi: mac80211: increment tx stats in tx encapsulation path

Tx packet count does not get updated in ifconfig utility output when
tx encapsulation is enabled in mac80211.

Fix this by incrementing tx stats in the tx encapsulation path.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 net/mac80211/tx.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/net/mac80211/tx.c b/net/mac80211/tx.c
index aca9e8c..e2ac14a 100644
--- a/net/mac80211/tx.c
+++ b/net/mac80211/tx.c
@@ -4877,6 +4877,8 @@ void ieee80211_8023_xmit_ap(struct ieee80211_sub_if_data *sdata,
 		atomic_inc(&sta->tx_netif_pkts);
 	}
 
+	ieee80211_tx_stats(dev, skb->len);
+
 	spin_lock_irqsave(&local->queue_stop_reason_lock, flags);
 
 	if (local->queue_stop_reasons[q] || !skb_queue_empty(&local->pending[q])) {
@@ -4926,11 +4928,13 @@ netdev_tx_t ieee80211_subif_start_xmit_8023(struct sk_buff *skb,
 				pubsta = &sta->sta;
 
 			control.sta = pubsta;
+			ieee80211_tx_stats(dev, skb->len);
 			drv_tx(sdata->local, &control,  skb);
 out:
 			rcu_read_unlock();
 		} else {
 			control.sta = NULL;
+			ieee80211_tx_stats(dev, skb->len);
 			drv_tx(sdata->local, &control,  skb);
 		}
 
-- 
2.34.1

